src/library.js

```diff
@@ -2388,6 +2388,23 @@ addToLibrary({
   _emscripten_get_now_is_monotonic__deps: ['$nowIsMonotonic'],
   _emscripten_get_now_is_monotonic: () => nowIsMonotonic,
 
+  __emscripten_atomics_sleep__internal: true,
+  // In a browser without cross origin isolation, SharedArrayBuffer is deleted
+  // from the global scope:
+  // https://html.spec.whatwg.org/multipage/webappapis.html#creating-a-new-javascript-realm
+  // However, it says:
+  //
+  // > Web developers can still get at the constructor through
+  // > `new WebAssembly.Memory({ shared:true, initial:0, maximum:0}).buffer.constructor`.
+  //
+  // That is what we are up to here. Use an IIFE to avoid shadowing
+  // SharedArrayBuffer globally.
+  __emscripten_atomics_sleep__postset: `
+    var SABConstructor = new WebAssembly.Memory({"shared":true,"initial":0,"maximum":0}).buffer.constructor;
+    var waitBuffer = new Int32Array(new SABConstructor(4));
+  `,
+  __emscripten_atomics_sleep: (ms) => Atomics.wait(waitBuffer, 0, 0, ms),
```

Comment:
Can the above code all be gated behind `#if !SHARED_MEMORY`?

Reply:
Yes that makes sense. I added a gate.